pkg/sbom/node.go

```diff
@@ -370,3 +383,12 @@ func (n *Node) HashesMatch(th map[int32]string) bool {
 	}
 	return atLeastOneMatch
 }
+
+// AddHash adds a new hash of algorithm algo to the node. If the node
+// already has a has of the same algorithm it will get silently replaced.
+func (n *Node) AddHash(algo HashAlgorithm, value string) {
+	if value == "" {
+		return
+	}
+	n.Hashes[int32(algo)] = value
```

Comment:
Howabout 
```
	if value == "" {
		return
	}
        if n.Hashes == nil {
		n.Hashes = map[int32]string{}
	}
        n.Hashes[int32(algo)] = value
```

Reply:
Very nice catch. I added your suggestion to the commit